lib/pka_mem.c

```diff
@@ -577,8 +581,9 @@ uint16_t pka_mem_alloc(uint32_t ring_id, uint32_t size)
     pka_mem_idx_t    chunk_idx;
     pka_mem_chunk_t *chunk;
     pka_mem_desc_t  *data_mem;
-    uint16_t         offset;
+    uint16_t         offset = 0;
 
+    pka_mem_lock();

```

Comment:
Does the code need to check the return code of the pka_mem_lock() call? In case of race condition.

Reply:
Should not be race condition here. Lock will block and wait until it’s release by other thread.